experiment/evaluator.py

```diff
@@ -195,13 +201,81 @@ def check_target(self, ai_binary, target_path: str) -> Optional[Result]:
       traceback.print_exc()
       return None
 
-  def _parse_libfuzzer_logs(self, log_handle) -> tuple[int, int, bool]:
+  def _parse_stacks_from_libfuzzer_logs(self, lines) -> list[str]:
+    """Parse stack traces from libFuzzer logs."""
+    # there may have more than 1 thread stack in a log
+    stacks = []
+
+    stack = []
+    for raw_line in lines:
+      line = raw_line.strip()
+
+      # stacks are continuous lines starting with '    #'
+      if not raw_line.startswith('#'):
+        if line.startswith('#'):
+          stack.append(line)
+          continue
+
+        if len(stack) > 0:
+          stacks.append(stack)
+          stack = []
+
+    if len(stack) > 0:
+      stacks.append(stack)
+
+    return stacks
+
+  def _parse_fuzz_cov_info_from_libfuzzer_logs(
+      self, lines) -> tuple[Optional[int], Optional[int], Optional[int]]:
+    """Parse cov of INITED & DONE, and round number from libFuzzer logs."""
+    initcov, donecov, lastround = None, None, None
+
+    for line in lines:
+      if line.startswith('#'):
+        # parse line starting with #int to get the round number
+        roundno = None
+        for ch in line:
+          if ch.isspace():
+            roundstr = line[1:line.index(ch)]
+            roundno = int(roundstr) if roundstr.isdigit() else None
+            break
+
+        if roundno is not None:
+          lastround = roundno
+          if 'INITED' in line:
+            initcov = int(line.split('cov: ')[1].split(' ft:')[0])
+          elif 'DONE' in line:
+            donecov = int(line.split('cov: ')[1].split(' ft:')[0])
+
+    return initcov, donecov, lastround
+
+  def _stack_func_is_of_testing_project(self, stack_func: str) -> bool:
+    if not CRASH_STACK_WITH_SOURCE_INFO.match(stack_func):
+      # stack_func of linked binary with no source info
+      # TODO: check this?
+      return False
+    if LLVM_SOURCE_PATH_PREFIX in stack_func:
+      return False
+    return True
+
+  def _parse_libfuzzer_logs(self,
+                            log_handle) -> tuple[int, int, bool, bool, str]:
     """Parse libFuzzer logs."""
+    lines = None
+    try:
+      fuzzlog = log_handle.read(-1)
+      # some driver crashes can mess up the libfuzzer output and raise exp
+      fuzzlog = fuzzlog.decode('utf-8', errors='ignore')
+      lines = fuzzlog.split('\n')
+    except MemoryError as _:
+      # some logs from abnormal drivers are too large to be parsed
+      return 0, 0, False, True, 'LOG_MESS_UP'
```

Comment:
Could you please log this so that we can be aware of it? E.g.,
```python
    except MemoryError as e:
      # some logs from abnormal drivers are too large to be parsed
     logging.waring('%s is too large to parse: %s', log_handle.name, e)
      return 0, 0, False, True, 'LOG_MESS_UP'
```